clang/test/CIR/CodeGen/loop.cpp

```diff
@@ -265,3 +265,125 @@ void test_empty_while_true() {
 // OGCG:   br label %[[WHILE_BODY:.*]]
 // OGCG: [[WHILE_BODY]]:
 // OGCG:   ret void
+
+void unreachable_after_continue() {
+  for (;;) {
+    continue;
+    int x = 1;
+  }
+}
+
+// CIR: cir.func @unreachable_after_continue
+// CIR:   cir.scope {
+// CIR:     cir.for : cond {
+// CIR:       %[[TRUE:.*]] = cir.const #true
+// CIR:       cir.condition(%[[TRUE]])
+// CIR:     } body {
+// CIR:       cir.scope {
+// CIR:         %[[X:.*]] = cir.alloca !s32i, !cir.ptr<!s32i>, ["x", init] {alignment = 4 : i64}
+// CIR:         cir.continue
+// CIR:       ^bb1:  // no predecessors
+// CIR:         %[[ONE:.*]] = cir.const #cir.int<1> : !s32i
+// CIR:         cir.store %[[ONE]], %[[X]] : !s32i, !cir.ptr<!s32i>
+// CIR:         cir.yield
+// CIR:       }
+// CIR:       cir.yield
+// CIR:     } step {
+// CIR:       cir.yield
+// CIR:     }
+// CIR:   }
+// CIR:   cir.return
+// CIR: }
+
+// LLVM: define void @unreachable_after_continue()
```

Comment:
What is going on here?  This seems like some strange unroll happening here, right?

Reply:
Can you run this and paste the CIR and outputted LLVM-IR here for me?  I'm having a hard time visualizing from what you said.

Reply:
> but I think right now that gets run before CIR flattening. That's something we should revisit.

Agreed, we should probably run it again after flattening / right before lowering. Note that LLVM optimizers are really good at simplifying these, so we haven't cared much enough just yet since even -O1 will do an amazing job here.

Reply:
https://godbolt.org/z/dj67e1dME

Reply:
Oh, ooof that is hideous.  But looks like it works at least.